docs/source/get_started/implementers_best_practices.rst

```diff
@@ -0,0 +1,135 @@
+recommend that the instance running the MOTECH server has got at least 4GB of RAM. Lower amounts may considerably
+slow down the processing.
+
+Of course the best way to find out the optimal configuration is to perform some stress testing. There are a lot of tools
+that can simulate the traffic to the server. Running such stress tools on servers with various configurations will help to
+determine at which point adding more resources stops adding up to the amount of processed requests.
+
+
+Topology recommendations
+========================
+
+MOTECH can be run in a clustered environment. This means that you can run several MOTECH instances in order to distribute
+the workload across multiple processing units. Similarly, you can set up multiple ActiveMQ instances that handle MOTECH
+events. In a topology with several MOTECH instances, you can choose whether to have the event handled by one instance
+only or, what is a rare case, by all instances. The second option is usually only used to notify the instances about some
+changes that must be immediately reflected on the instances. Finally, it is also possible to have clustered databases.
+There are two approaches when it comes to clustering databases. In the master-master model, all of the databases can update
+the data and the changes get replicated across all of the instances. In the master-slave model, only one of the databases
+(considered master) handles data updates and pushes the updated data to the slave servers. Of course using clustered
+environment is not a requirement. You can use single MOTECH, ActiveMQ and database instance, depending on the expected
+traffic.
+
+
+Releases
+========
+
+Major MOTECH releases usually do not come periodically. The release date is dictated by the amount of tickets assigned
+to next versions and the time it takes to resolve them. Each new major release is announced on our `mailing list <https://groups.google.com/forum/?fromgroups#!forum/motech-dev>`_.
+We strongly recommend using stable MOTECH versions.
+
+In case the MOTECH version your implementation uses contains some bug, that seriously impacts your implementation,
+feel free to contact us. We will see if we can deliver the bug fix in form of a point release.
+
+
+Building modules
+================
+
+Building implementations is often connected with the need to create additional modules, that will work with MOTECH.
+To ease and speed up the process of building a new module, we have prepared a set of maven archetypes, that help
+to start developing the module. In simple words, archetypes are "Hello World" MOTECH modules, containing all the
+basic configuration and files. You can pick a minimal archetype, that will contain only files and code required to have
+a module, or add more features, like module UI and controllers, communication with the database and module configuration
+and settings. If you plan to develop modules during the implementation, we highly recommend using our archetypes. You
```

Comment:
for your implementation. Anyway, doesn't it being an implementation mean it will have its own modules? If not, wouldn't that just be a deployment? hmm